### ISSUE TITLE
Possible feature

### ISSUE THREAD

**Issue by ManoogD** [explooosion/Agm-Direction#13]
Hi. We're using your project and it works find, thanks.
But there is a one feature which we need it's a calculate distance of direction and I don't know it is possible to implement somehow with your project or not. If it's not, maybe you know how it could be achieve? Thank you.

**Comment by ManoogD** [comment]
@explooosion  Thanks!

**Comment by lolaswift** [comment]
I second this. It would be really awesome
@explooosion I am also using your project in production. thx again!

**Comment by explooosion** [comment]
Hi! @ManoogD .
There are so many fields, so I emitted all response.

I added the `@Output` - onChange.
Please install the latest version `0.5.0`.

- fields : [DirectionsResult](https://developers.google.com/maps/documentation/javascript/directions?hl=en#DirectionsResults)

### Example

Here is an example to `The DirectionsResult Object`

HTML

```html
<agm-direction ... (onChange)="change($event)"></agm-direction>
```

TS

```ts
public change(e) {

  // You can do anything here.
  console.log(e)

  /**
   * Example: Get the distance of direction
   */
  const legs = e.routes[0].legs
  let disTotal = 0
  legs.forEach(value => {
    disTotal += value.distance.value
  })
  console.log(`distance: ${disTotal}m`)
  // distance: 3100m (example)

  /**
   * Example: Get all points of the direction passed
   */
  const overview_path = e.routes[0].overview_path
  let passed = []
  overview_path.forEach(value => {
    passed.push({
      lat: value.lat(),
      lng: value.lng(),
    })
  })
  console.log(passed)
  // [{ lat:x1, lng:y1 },{ lat:x2, lng:y2 }, ...]

}
```

**Comment by ManoogD** [comment]
@explooosion Thank you so much! You're AWESOME!

**Comment by lolaswift** [comment]
@explooosion The npm package is still at version 0.4.3. Can you please update it to the latest version?

**Comment by explooosion** [comment]
Hi @lolaswift !
Please try to remove the local package.
```bash
npm uninstall agm-direction
```

Or install the latest version.
```bash
npm install --save agm-direction@latest
```

You can also specify version `0.5.1`
```bash
npm install --save agm-direction@0.5.1
```

**Comment by explooosion** [comment]
This issue was released new version ( [0.5.1](https://github.com/explooosion/Agm-Direction/releases/tag/0.5.1) ).
